pkg/controller/kieapp/webconsole.go

```diff
@@ -0,0 +1,141 @@
+package kieapp
+
+//go:generate go run -mod=vendor ./defaults/.packr/packr.go
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/ghodss/yaml"
+	"github.com/gobuffalo/packr/v2"
+	"strconv"
+	"strings"
+
+	consolev1 "github.com/openshift/api/console/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	TargetNamespace   = "openshift-console"
+	ConsoleYAMLSample = "ConsoleYAMLSample"
+)
+
+type CustomResourceDefinition struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+}
+
+func CreateConsoleYAMLSamples(r *Reconciler) {
+	log.Info("Loading CR yaml samples.")
+	contents, names, err := loadFilesOnlyWithBox("cryamlsamples")
+	if err != nil {
+		log.Error(err, "Failed to load cryamlsamples")
+		return
+	}
+
+	resMap, err := applyMultipleWebConsoleYamls(contents, r)
+	if err != nil {
+		log.Error(err, "cryamlsamples not successfully applied")
+		return
+	}
+
+	for k, v := range resMap {
+		log.Info("yaml file: %s %s ", names[k], v)
+	}
+}
+
+func applyMultipleWebConsoleYamls(yamlString []string, r *Reconciler) (map[int]string, error) {
+	if yamlString == nil || len(yamlString) == 0 {
+		return nil, errors.New("Empty yaml list")
+	}
+	resMap := make(map[int]string)
+	for idx, yamlStr := range yamlString {
+		_, err := create(yamlStr, r)
+		if err != nil {
+			resMap[idx] = err.Error()
+		} else {
+			resMap[idx] = "Applied"
+		}
+	}
+	return resMap, nil
+}
+
+func create(yamlStr string, r *Reconciler) (bool, error) {
+	obj := &CustomResourceDefinition{}
+	err := yaml.Unmarshal([]byte(yamlStr), obj)
```

Comment:
Our unit tests or somewhere should have the right way of reading our CR files

Reply:
modified